### PR TITLE
Dashboard glitch fix

### PR DESCRIPTION
**Pull request by Lrthayer**

### CODE REVIEW

File: app/views/dashboard.pug [hunk 1]
```diff
@@ -2,20 +2,21 @@ doctype html
...
         link(rel="stylesheet", href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap-theme.min.css")
+        link(rel="stylesheet", src ="/externalCSS.css")
         script(src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js")
+        script(type="text/javascript" src="//ajax.googleapis.com/ajax/libs/jquery/2.0.0/jquery.min.js")
 
-    body(style='width:100%;height:100%;background-image:url("/assets/background.png");')
+    body(id='full' style='background-image:url("/assets/background.png");')
 
-        div.container(style='width:100%;height:100%;')
+        div.container(id='full')
```
Comment (DannyPeck) [inline app/views/dashboard.pug]: I don't think we need this, if we do please use classes instead of id's.

File: app/views/dashboard.pug [hunk 2]
```diff
@@ -2,20 +2,21 @@ doctype html
...
 
-            div(id='topHalf' style='float:top;height:20%;')
+            div(id='topFloat')
```
Comment (DannyPeck) [inline app/views/dashboard.pug:15]: Try having this title not touch the left side of the page so the button bar can span the whole left side.

File: app/views/dashboard.pug [hunk 3]
```diff
@@ -2,20 +2,21 @@ doctype html
...
                 div(id='mainTitle' style='text-align:center;')
                     h1 Welcome to Hive!
 
-            div(id='bottomHalf' style='position:absolute;width:100%;height:80%;left:0%;top:20%;')
+            div(id='bottomFloat2' style='position:absolute;width:100%;height:80%;left:0%;top:20%;')
```
Comment (DannyPeck) [inline app/views/dashboard.pug]: Is there a bottomFloat1?

File: app/views/dashboard.pug [hunk 4]
```diff
@@ -38,7 +39,9 @@ html(lang="en")
                         div(style='position:absolute;width:100%;height:20%;left:0%;top:80%;')
                             button.btn.btn-default(type='submit' style='width:100%;height:100%;left:0%;top:0%;') Logout
 
-                div(id='mainDiv' style='position:absolute;width:90%;height100%;left:10%;top:0%;border-style:solid;')
+                div(id='mainDiv' style='position:absolute;width:90%;height100%;left:10%;top:0%;border-style:solid; z-index:1')
```
Comment (DannyPeck) [inline app/views/dashboard.pug]: What is the purpose of mainDiv and why do we need to hide it?

**Review by DannyPeck — CHANGES_REQUESTED** [review]
A definite improvement, just a few changes and design questions!

File: app/views/dashboard.pug (hunk 1 above)
Comment (Lrthayer) [reply app/views/dashboard.pug]: I believe you are correct, and noted.

File: app/views/dashboard.pug (hunk 3 above)
Comment (Lrthayer) [reply app/views/dashboard.pug]: nope that is just classic lazy naming conventions, I didn't know if there was a bottomFloat1 so I made 2 for testing and forgot to fix it. I will fix that.

Comment (DannyPeck) [inline app/views/dashboard.pug]: If not just change it to bottomFloat

File: app/views/dashboard.pug (hunk 4 above)
Comment (Lrthayer) [reply app/views/dashboard.pug]: mainDiv is suppose to be the div we reuse for every tab that doesn't use a pop-up. I think the idea was that the pop-up would overlay the mainDiv at all times but with the glitches that didn't work out. That is how I fixed the glitch, is I hide the main div whenever a pop-up is up.

Comment (DannyPeck) [reply app/views/dashboard.pug]: Ok, what glitches occurred when we tried to overlay the mainDiv with the pop up?

Comment (Lrthayer) [reply app/views/dashboard.pug]: That was the weird text stuff. The pop up would be visible but the mainDiv's text was also slightly visible.

File: app/views/dashboard.pug [hunk 5]
```diff
@@ -2,20 +2,21 @@ doctype html
...
         title User Dashboard
-        script (src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js")
         link(rel="stylesheet", href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css")
         link(rel="stylesheet", href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap-theme.min.css")
+        link(rel="stylesheet", src ="/externalCSS.css")
         script(src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js")
+        script(type="text/javascript" src="//ajax.googleapis.com/ajax/libs/jquery/2.0.0/jquery.min.js")
 
-    body(style='width:100%;height:100%;background-image:url("/assets/background.png");')
+    body(id='full' style='background-image:url("/assets/background.png");')
```
Comment (DannyPeck) [inline app/views/dashboard.pug]: I think this also needs to be changed to say class='full'

File: app/views/dashboard.pug [hunk 6]
```diff
@@ -108,3 +115,11 @@ html(lang="en")
                         div.container(name='composeSend' style='position:absolute;width:100%;height:5%;left:0%;top:95%;')
                             form.form-horizontal(method='post', action='/dashboard/compose/send')
                                 button.btn.btn-default(type='submit' style='position:absolute;width:20%;height:100%;left:80%;top:0%;') Send
+            else if (notificationVisible === 'true')
+                script.
+                    $('#mainDiv').show()
+            else
+                script.
+                    console.log("?")
```
Comment (DannyPeck) [inline app/views/dashboard.pug]: We should remove this console.log statement.

File: app/views/dashboard.pug [hunk 7]
```diff
@@ -2,20 +2,21 @@ doctype html
...
 
-    body(style='width:100%;height:100%;background-image:url("/assets/background.png");')
+    body(id='full' style='background-image:url("/assets/background.png");')
 
-        div.container(style='width:100%;height:100%;')
+        div.container(class='full')
 
-            div(id='topHalf' style='float:top;height:20%;')
-                div(id='mainTitle' style='text-align:center;')
+            div(id='topFloat')
```
Comment (DannyPeck) [inline app/views/dashboard.pug:15]: So, I think it is now 80% but still blocking the left side. Not sure exactly how that works.  It's not a big deal really, we can handle that in a different PR. Don't worry about it here.

**Review by DannyPeck — CHANGES_REQUESTED** [review]
Coming along, just a few more small changes and we should be good to go.

File: app/views/dashboard.pug (hunk 7 above)
Comment (Lrthayer) [reply app/views/dashboard.pug:15]: yeah sorry, took a shot in the dark tbh. I will work with that more carefully next time

**Review by DannyPeck — APPROVED** [review]
Ok, I think this a really good step in the right direction!

**Pull request merged**